webapp/src/test/java/com/box/l10n/mojito/service/assetintegritychecker/integritychecker/PrintfLikeIgnorePercentageAfterBracketsIntegrityCheckerTest.java

```diff
@@ -0,0 +1,296 @@
+        }
+    }
+
+    @Test
+    public void testAndroidPlaceholderCheckFailsIfSamePlaceholdersCountButSpecifierModified() throws PrintfLikeWhitespaceInTokenInvalidIntegrityCheckerException {
+
+        PrintfLikeIgnorePercentageAfterBracketsIntegrityChecker checker = new PrintfLikeIgnorePercentageAfterBracketsIntegrityChecker();
+        String source = "There are %1$d files and %2$d folders";
+        String target = "Il y a %1$d fichiers et %2$s dossiers";
+
+        try {
+            checker.check(source, target);
+            fail("PrintfLikeWhitespaceInTokenInvalidIntegrityCheckerException must be thrown");
+        } catch (PrintfLikeWhitespaceInTokenInvalidIntegrityCheckerException e) {
+            assertEquals(e.getMessage(), "Placeholders in source and target are different");
+        }
+    }
+
+    @Test
+    public void testSpecifierWithRemovedSpace() throws PrintfLikeWhitespaceInTokenInvalidIntegrityCheckerException {
+
+        PrintfLikeIgnorePercentageAfterBracketsIntegrityChecker checker = new PrintfLikeIgnorePercentageAfterBracketsIntegrityChecker();
+        String source = "%1$s of %2$s GB";
+        String target = "%1$s/%2$sGB";
+
+        checker.check(source, target);
+    }
+
+    @Test
+    public void testIncorrectlyModifiedSpecifier() throws PrintfLikeWhitespaceInTokenInvalidIntegrityCheckerException {
+
+        PrintfLikeIgnorePercentageAfterBracketsIntegrityChecker checker = new PrintfLikeIgnorePercentageAfterBracketsIntegrityChecker();
+        String source = "%1.1f GB";
+        String target = "%1,1f Go";
+
+        try {
+            checker.check(source, target);
+            fail("PrintfLikeWhitespaceInTokenInvalidIntegrityCheckerException must be thrown");
+        } catch (PrintfLikeWhitespaceInTokenInvalidIntegrityCheckerException e) {
+            assertEquals(e.getMessage(), "Placeholders in source and target are different");
+        }
+    }
+
+    @Test
+    public void testTokenIsNotRecognizedIfPrecededByRightCurlyBracket() throws PrintfLikeWhitespaceInTokenInvalidIntegrityCheckerException {
+        // Verify that "% to" is not matched as a token
+        PrintfLikeIgnorePercentageAfterBracketsIntegrityChecker checker = new PrintfLikeIgnorePercentageAfterBracketsIntegrityChecker();
+        String source = "from {from_val}% to {to_val}%";
+        String target = "{from_val}%-ról {to_val}%-ra";
+
+        checker.check(source, target);
+    }
+
+    @Test
+    public void testTokenIsNotRecognizedIfPrecededByRightBracket() throws PrintfLikeWhitespaceInTokenInvalidIntegrityCheckerException {
+        // Verify that "% to" is not matched as a token
+        PrintfLikeIgnorePercentageAfterBracketsIntegrityChecker checker = new PrintfLikeIgnorePercentageAfterBracketsIntegrityChecker();
+        String source = "from (from_val)% to (to_val)%";
+        String target = "(from_val)%-ról (to_val)%-ra";
+
+        checker.check(source, target);
+    }
+
```

Comment:
Mark, first, thanks for adding this checker!

Please also add unit tests for the following:

- negative tests where % is used as a plain old percent sign. eg. "The level is down to 10% of the original." should be an error because the percent char is not doubled.
- tests where the % is doubled to make an actual percent sign appear in the output. eg. "The level is down to 10%% of the original." should not be an error.
- a test in Turkish. My projects often fail on Turkish because they format percentages where the percent symbol comes first: "The level is down to 10% of the original." -> "Seviye, orijinalin %10'una düştü." should not be an error (but often is with the other integrity checkers)

Reply:
Hi Edwin,
Thanks for the comments, I've pushed an update that adds the above tests. I've made the assumption that for the turkish test the percentages should be doubled i.e. "The level is down to 10%% of the original." -> "Seviye, orijinalin %%10'una düştü."

Reply:
Yes, good point about the Turkish test! Thanks for adding those extra tests